### PR TITLE
Add let* and let+ definitions to Fiber

### PR DESCRIPTION
**Pull request by rgrinberg**
Use them instead of >>= and >>| where appropriate

### CODE REVIEW

File: bin/import.ml [hunk 1]
```diff
@@ -54,9 +53,9 @@ module Main = struct
 
   let setup ~log ?external_lib_deps_mode (common : Common.t) =
     scan_workspace ~log common
-    >>= init_build_system
-          ?external_lib_deps_mode
-          ?only_packages:common.only_packages
+    |> Fiber.bind ~f:(init_build_system
```
Comment (diml) [inline bin/import.ml]: Why this change? I find the original code simpler

**Comment by diml** [comment]
That seems good, but I wonder if we should introduce two let-operators just for fibers. What about introducing just one for `bind` and using `Fiber.return` more often?

We might want to use distinct let-operators between the various monads/applicative in dune BTW. It'll make it easier to remember what is what.

File: bin/import.ml (hunk 1 above)
Comment (rgrinberg) [reply bin/import.ml]: I'll revert it. It's because I don't see `>>=` and `>>|` to be very useful anymore.

**Comment by rgrinberg** [comment]
> We might want to use distinct let-operators between the various monads/applicative in dune BTW. It'll make it easier to remember what is what.

I think it's a bit difficult to remember that we left off the applicative for _some_ monads. I can do this if other people are in favor.

File: bin/import.ml (hunk 1 above)
Comment (rgrinberg) [reply bin/import.ml]: And also because we now don't need an `open`.

Comment (diml) [reply bin/import.ml]: If you want to remove the `>>=`, I would suggest to just use `let*` for consistency. Introducing intermediate variable often helps readability anyway.

**Comment by diml** [comment]
Well, maybe we don't need to do this eagerly. These operators are quite new and we'll need a bit of experience working with them to see what work best.
